### PR TITLE
Modernize Plugin

### PR DESCRIPTION
**Pull request by janasrikanth**
### Testing done

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -5,26 +5,19 @@
...
     <groupId>org.jenkins-ci.plugins</groupId>
     <artifactId>plugin</artifactId>
-    <version>2.11</version>
+    <version>4.80</version>
     <relativePath />
   </parent>
   <groupId>org.jenkins-ci.plugins</groupId>
   <artifactId>device-watcher</artifactId>
-  <version>0.10-SNAPSHOT</version>
+  <version>${revision}${changelist}</version>
```
Comment (NotMyFault) [inline pom.xml]:
```suggestion
  <version>${changelist}</version>
```

File: pom.xml [hunk 2]
```diff
@@ -5,26 +5,19 @@
...
-    <java.level>7</java.level>
-    <!-- Jenkins Test Harness version you use to test the plugin. -->
-    <!-- For Jenkins version >= 1.580.1 use JTH 2.x or higher. -->
-    <jenkins-test-harness.version>2.13</jenkins-test-harness.version>
-    <!-- Other properties you may want to use:
-         ~ hpi-plugin.version: The HPI Maven Plugin version used by the plugin..
-         ~ stapler-plugin.version: The Stapler Maven plugin version required by the plugin.
-    -->
+    <revision>0.10</revision>
+    <changelist>-SNAPSHOT</changelist>
```
Comment (NotMyFault) [inline pom.xml]: <changelist>999999-SNAPSHOT</changelist>
```

File: pom.xml [hunk 3]
```diff
@@ -49,9 +42,9 @@
   </developers>
   <!-- Assuming you want to host on @jenkinsci:-->
   <scm>
-    <connection>scm:git:git://github.com/jenkinsci/device-watcher-plugin.git</connection>
+    <connection>scm:git:https://github.com/${gitHubRepo}.git</connection>
     <developerConnection>scm:git:git@github.com:jenkinsci/device-watcher-plugin.git</developerConnection>
-    <url>http://github.com/jenkinsci/device-watcher-plugin</url>
+    <url>https://github.com/jenkinsci/device-watcher-plugin</url>
     <tag>device-watcher-0.9</tag>
```
Comment (NotMyFault) [inline pom.xml]:
```suggestion
    <tag>${scmTag}</tag>
```

File: pom.xml [hunk 4]
```diff
@@ -5,26 +5,19 @@
...
-    <!-- Jenkins Test Harness version you use to test the plugin. -->
-    <!-- For Jenkins version >= 1.580.1 use JTH 2.x or higher. -->
-    <jenkins-test-harness.version>2.13</jenkins-test-harness.version>
-    <!-- Other properties you may want to use:
-         ~ hpi-plugin.version: The HPI Maven Plugin version used by the plugin..
-         ~ stapler-plugin.version: The Stapler Maven plugin version required by the plugin.
-    -->
+    <revision>0.10</revision>
+    <changelist>-SNAPSHOT</changelist>
+    <jenkins.version>2.361.4</jenkins.version>
```
Comment (NotMyFault) [inline pom.xml]: This should adhere with the baseline recommendations, if you are modernizing the plugin, as outlined in the PR title.

**Review by NotMyFault — CHANGES_REQUESTED** [review]
The .mvn folder is missing too.

File: pom.xml (hunk 4 above)
Comment (janasrikanth) [reply pom.xml]: Thanks for reviewing @NotMyFault 
Added suggested changes, can you please check

**Comment by janasrikanth** [comment]
Hi @NotMyFault 
I would like to take your review before merging. can you please check

**Comment by janasrikanth** [comment]
Hi @NotMyFault 
I hope this PR has no issues to merge

**Comment by NotMyFault** [comment]
I am not a maintainer of this plugin, no need to ping me.

**Pull request merged**